namui/src/namui/render/rendering_tree.rs

```diff
@@ -524,18 +525,167 @@ impl RenderingTree {
+                            [1.0, 0.0, translate.x],
+                            [0.0, 1.0, translate.y],
+                            [0.0, 0.0, 1.0],
+                        ]);
+                        let matrix = translation_matrix * matrix;
+                        get_bounding_box_with_matrix_of_rendering_trees(
+                            translate.rendering_tree.iter(),
+                            &matrix,
+                        )
+                    }
+                    SpecialRenderingNode::Clip(clip) => {
+                        get_bounding_box_with_matrix_of_rendering_trees(
+                            clip.rendering_tree.iter(),
+                            &matrix,
+                        )
+                        .and_then(|bounding_box| {
+                            let path = clip.path_builder.build();
+
+                            let clip_bounding_box = path.get_bounding_box();
+
+                            match clip.clip_op {
+                                ClipOp::Intersect => {
+                                    clip_bounding_box.and_then(|clip_bounding_box| {
+                                        bounding_box.intersect(&clip_bounding_box)
+                                    })
+                                }
+                                ClipOp::Difference => match clip_bounding_box {
+                                    Some(clip_bounding_box) => {
+                                        if bounding_box == clip_bounding_box {
+                                            return None;
+                                        }
+
+                                        let xs = [
+                                            bounding_box.left,
+                                            bounding_box.right,
+                                            clip_bounding_box.left,
+                                            clip_bounding_box.right,
+                                        ];
+                                        let ys = [
+                                            bounding_box.top,
+                                            bounding_box.bottom,
+                                            clip_bounding_box.top,
+                                            clip_bounding_box.bottom,
+                                        ];
+
+                                        let sixteen_xys = xs
+                                            .iter()
+                                            .zip(ys.iter())
+                                            .map(|(x, y)| Xy { x: *x, y: *y });
+
+                                        let difference_area_xys = sixteen_xys.filter(|xy| {
+                                            (clip_bounding_box.is_xy_outside(&xy)
+                                                || clip_bounding_box.is_xy_on_border(&xy))
+                                                && !bounding_box.is_xy_outside(&xy)
+                                        });
```

Comment:
Can you get what I did?

Reply:
At first, there was a bit of confusion because I thought of the LTRB of css. But it's okay now.

![image](https://user-images.githubusercontent.com/38313680/167811765-a7172ce9-2594-4afe-9f8a-b90232e5aab2.png)